lib/elixir/lib/file.ex

```diff
@@ -1388,4 +1393,18 @@ defmodule File do
     do: path
   defp maybe_to_string(path),
     do: IO.chardata_to_string(path)
+
+  @doc false
+  def bom_length(<<239, 187, 191, _rest::binary>>),
```

Comment:
I decided to place it here, because there may be other places that may use this function, for example some `IO/File.read` fun may also benefit from `:trim_bom` mode. But thats my bad, "premature refactoring".. :)